### PR TITLE
fix(core): add warning if running on an outdated global install

### PR DESCRIPTION
**Pull request by AgentEnder**
Reapplies (#15442)

This reverts commit 1eee5f0.

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by barbados-clemens — APPROVED** [review]

File: e2e/utils/create-project-utils.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ export function newProject({
 
       if (useBackupProject) {
         // stop the daemon
-        execSync('nx reset', {
+        execSync(`${getPackageManagerCommand().runNx} reset`, {
```
Comment (meeroslav) [inline e2e/utils/create-project-utils.ts:85]: Good catch

File: packages/nx/bin/nx.ts [hunk 2]
```diff
@@ -93,3 +102,61 @@ function resolveNx(workspace: WorkspaceTypeAndRoot | null) {
...
+    globalVersion &&
+    ((localNxVersion && major(globalVersion) < major(localNxVersion)) ||
+      (!localNxVersion &&
+        getLatestVersionOfNx() &&
+        major(globalVersion) < major(getLatestVersionOfNx())));
+
+  // Using a global Nx Install
+  if (isOutdatedGlobalInstall) {
+    const bodyLines = localNxVersion
+      ? ['Its time to update Nx 🎉']
```
Comment (meeroslav) [inline packages/nx/bin/nx.ts:122]: This would show `Its time to update Nx 🎉` twice - once as a title and once as the first line in the body.

**Review by mandarini — APPROVED** [review]

**Review by mandarini — APPROVED** [review]

File: e2e/utils/create-project-utils.ts [hunk 3]
```diff
@@ -81,7 +81,7 @@ export function newProject({
       packageInstall(packages.join(` `), projScope);
 
       // stop the daemon
-      execSync('nx reset', {
+      execSync(`${getPackageManagerCommand().runNx} reset`, {
```
Comment (meeroslav) [inline e2e/utils/create-project-utils.ts:84]: This should be already addressed by another merged PR, but good catch regardless

File: docs/shared/guides/global-nx.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+With a global Nx installation, Nx looks for the local copy of Nx in your repo and hands off the process execution to it. This means that whichever version of Nx is installed locally in your repo is still the version of Nx that runs your code. For the most part, this can eliminate any issues that may arise from the global install being outdated.
+
+However, there are still cases where an issue could arise. If the structure of your Nx workspace no longer matches up with what the globally installed copy of Nx expects, it may fail to hand off to your local installation properly and instead error. This commonly results in errors such as:
+
+- Could not find Nx modules in this workspace.
+- The current directory isn't part of an Nx workspace.
+
+If you find yourself in this position, you will need to update your global install of Nx.
+
+## Updating your global Nx installation
```
Comment (meeroslav) [inline docs/shared/guides/global-nx.md:18]: Let's call this `Installing your global Nx installation` and move after the paragraph on line 9.

We can then refer to it for update section and later when we remove the packages.

**Review by meeroslav — APPROVED** [review]
Left some comments about the guide.

The rest looks good, not sure why it fails on CI.

File: e2e/utils/create-project-utils.ts (hunk 3 above)
Comment (AgentEnder) [reply e2e/utils/create-project-utils.ts:84]: This PRs rebased and still needs the change. 🤷

File: nx.json [hunk 5]
```diff
@@ -120,6 +120,9 @@
         },
         {
           "env": "NX_E2E_CI_CACHE_KEY"
+        },
+        {
+          "env": "CI"
```
[the remark this answers is not in the document]
Comment (AgentEnder) [reply nx.json:125]: Prevents cache hits when trying to recreate CI runs locally (e.g. `CI=true nx e2e e2e-nx-misc...`

**Comment by AgentEnder** [comment]
> Left some comments about the guide.
> 
> The rest looks good, not sure why it fails on CI.

I've not figured out why it fails on CI yet either, locally it passes.... I'm wondering if the global install isn't being picked up properly.

**Review by bcabanes — APPROVED** [review]

**Pull request merged**